### PR TITLE
Updating the build documentation with instructions to build mxnet with NCCL

### PR DESCRIPTION
**Pull request by leleamol**
## Description ##
With this change, I have added the steps to build and validate MXNet with NCCL libraries.

### Essentials ###
- [y] Passed code style checking (`make lint`)
- [y] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [y] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by leleamol** [comment]
@eric-haibin-lin  @ptrendx  This is a fresh PR that replaces the PR https://github.com/apache/incubator-mxnet/pull/8957

File: docs/install/build_from_source.md [hunk 1]
```diff
@@ -317,6 +317,40 @@ Use [CMake](https://cmake.org/) to create a Visual Studio solution in ```./build
...
+make -j"$(nproc)"
+```
+
+## Validation
+- Follow the steps to install MXNet Python binding.
+- Comment the following line in ```test_nccl.py``` file at ```incubator-mxnet/tests/python/gpu/test_nccl.py```
+``` bash
+@unittest.skip("Test requires NCCL library installed and enabled during build")
+```
+- Run test_nccl.py script as follows. The test should complete. It does not produce any output.
```
Comment (eric-haibin-lin) [inline docs/install/build_from_source.md]: `Run test_nccl.py script as follows ` -> The command is still missing. Please add 
```
nosetests --verbose tests/python/gpu/test_nccl.py
```

Comment (leleamol) [reply docs/install/build_from_source.md]: Added the missing line.

**Comment by leleamol** [comment]
Thanks!

**Pull request merged**